business/workloads.go

```diff
@@ -1678,13 +1679,42 @@ func fetchWorkload(ctx context.Context, layer *Layer, criteria WorkloadCriteria)
 			}
 		}
 
+		// Check if there are any waypoint proxy binded to this service account
+		if w.IstioAmbient {
+			w.Waypoint = listWaypoint(ctx, layer, criteria.Namespace, w)
+		}
+
 		if cnFound {
 			return &w, nil
 		}
 	}
 	return wl, kubernetes.NewNotFound(criteria.WorkloadName, "Kiali", "Workload")
 }
 
+// Return the list of waypoint proxy binded to a workload service account
+func listWaypoint(ctx context.Context, layer *Layer, namespace string, w models.Workload) []string {
+	// Get List of waypoint proxies from the namespace and compare the
```

Comment:
`...and compare the` .... what does it compare? :) comment is incomplete.

Reply:
That's right, thanks for pointing at this! I have reworked that function (And completed the comment). I think that would be more useful to check, for a waypoint proxy, the list of applications with the same service account that the Waypoint proxy is attached to. So, we can show a bit more of information in the Waypoint Details view: 

![image](https://user-images.githubusercontent.com/49480155/230905705-81ab245c-1cf6-4b2b-8198-35c4192ab669.png)